packages/client/src/editor.tsx

```diff
@@ -37,6 +42,39 @@ export default function Editor() {
     new CodemirrorBinding(yText, editor, provider.awareness, {
       yUndoManager,
     });
-  }, [ref]);
+  }, [ref, docId]);
+
+  useEffect(() => {
+    fetch("/api/me")
+      .then((data) => data.json())
+      .then((data) => {
+        if (!data.user) return;
+        setMe(data.user);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!me) {
+      return;
+    }
```

Comment:
Are you declaring two useEffect functions, or inputting functions as a parameter to each? What is going on here?

Reply:
I am using two effects here with differing dependencies - the useEffect with `[]` as the dependency array restricts the effect to executing only whenever the Editor component gets mounted. In this effect I am requesting information about the currently logged in user via the "/api/me" route, and then setting `me`, a state variable for the Editor component. The other useEffect depends on `me` (dependency array of useEffect: `[me]`), so whenever the Editor component is mounted/when `me` changes value, the effect will run. In this effect, I am requesting a user doc via "/api/documents", or creating one if it doesn't exist via "api/document".